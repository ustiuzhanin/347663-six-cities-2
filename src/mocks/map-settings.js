export const mapSettings = {
  zoom: 12,
  icon: {
    url: `/img/pin.svg`,
    size: [30, 30]
  }
};
