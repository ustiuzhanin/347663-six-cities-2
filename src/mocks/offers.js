export const offers = [
  {
    id: 1,
    src: `img/room.jpg`,
    price: `80`,
    rating: `70`,
    name: `Beautiful & luxurious apartment at great location`,
    type: `Private room`
  },
  {
    id: 2,
    src: `img/room.jpg`,
    price: `120`,
    rating: `40`,
    name: `Wood and stone place`,
    type: `House`
  },
  {
    id: 3,
    src: `img/room.jpg`,
    price: `20`,
    rating: `100`,
    name: `Canal View Prinsengracht`,
    type: `Private room`
  },
  {
    id: 4,
    src: `img/room.jpg`,
    price: `60`,
    rating: `50`,
    name: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`
  }
];
