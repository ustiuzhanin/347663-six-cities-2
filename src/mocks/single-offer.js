export const offer = {
  city: {
    _id: "6005f36637751d35c8503011",
    name: `Dusseldorf`,
    location: { latitude: 51.225402, longitude: 6.776314, zoom: 13 },
  },
  preview_image: `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg`,
  images: [
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/5.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/19.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/1.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/2.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/10.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/13.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/20.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/4.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/8.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/11.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/16.jpg`,
  ],
  title: `Canal View Prinsengracht`,
  is_premium: false,
  rating: 3.1,
  type: `room`,
  bedrooms: 1,
  max_adults: 2,
  price: "101",
  goods: [`Breakfast`, `Laptop friendly workspace`],
  host: "6005f36637751d35c8503011",
  description: `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
  location: {
    latitude: 51.248402000000006,
    longitude: 6.763314,
    zoom: 16,
  },
  _id: "6005f36637751d35c8503011",
};
export const offerWithHost = {
  city: {
    _id: "6005f36637751d35c8503011",
    name: `Dusseldorf`,
    location: { latitude: 51.225402, longitude: 6.776314, zoom: 13 },
  },
  preview_image: `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg`,
  images: [
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/5.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/19.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/1.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/2.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/10.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/13.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/20.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/4.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/8.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/11.jpg`,
    `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/16.jpg`,
  ],
  title: `Canal View Prinsengracht`,
  is_premium: false,
  rating: 3.1,
  type: `room`,
  bedrooms: 1,
  max_adults: 2,
  price: "101",
  goods: [`Breakfast`, `Laptop friendly workspace`],
  host: {
    _id: "6005f36637751d35c8503011",
    avatar_url: "/img/avatar.svg",
    bookmarks: ["5fee3aaa54b627023d7f94b7"],
    email: "test1@test.com",
    is_pro: false,
    name: "test",
    offers: [],
  },
  description: `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
  location: {
    latitude: 51.248402000000006,
    longitude: 6.763314,
    zoom: 16,
  },
  _id: "6005f36637751d35c8503011",
};
