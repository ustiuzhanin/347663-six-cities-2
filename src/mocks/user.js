export const user = {
  _id: "6005f36637751d35c8503011",
  avatar_url: "/img/avatar.svg",
  bookmarks: ["5fee3aaa54b627023d7f94b7"],
  email: "test1@test.com",
  is_pro: false,
  name: "test",
  offers: [],
};
