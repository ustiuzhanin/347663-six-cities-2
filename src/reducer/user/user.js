const initialState = {
  user: {},
};

const ActionType = {
  GET_USER: `GET_USER`,
};

const ActionCreator = {
  getUser: (user) => {
    return {
      type: ActionType.GET_USER,
      payload: user,
    };
  },
};

const Operations = {
  getUser: (id) => (dispatch, getState, api) => {
    return api
      .get(`/get-user/${id}`)
      .then((response) => {
        if (response.status === 200) {
          console.log(response.data);
          dispatch(ActionCreator.getUser(response.data));
        }
      })
      .catch((err) => console.log(err));
  },
  changeBookmark: (id) => (dispatch, getState, api) => {
    const token = localStorage.token;
    return api
      .put(
        `/change-bookmark/${id}`,
        {},
        {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        }
      )
      .then((response) => {
        console.log(response);
      })
      .catch((err) => console.log(err));
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.GET_USER:
      return Object.assign({}, state, {
        user: action.payload,
      });
  }

  return state;
};

export { ActionCreator, ActionType, Operations, reducer };
