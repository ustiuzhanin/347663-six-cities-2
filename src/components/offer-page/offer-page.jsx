import React, { useEffect, useState } from "react";
import { connect } from "react-redux";
import { Operations } from "../../reducer/offers/offers";
import { Operations as UserOperations } from "../../reducer/user/user";
import { ActionCreator } from "../../reducer/auth/auth";

import PropTypes from "prop-types";

import Header from "../header/header.jsx";
import AuthModal from "../auth-modal/auth-modal.jsx";
// TODO: add comments
// import Comments from "../comments/comments.jsx";
// import CommentForm from "../comment-form/comment-form.jsx";
import Map from "../map/map.jsx";
import Card from "../card/card.jsx";

const OfferPage = (props) => {
  const {
    isAuthorizationRequired,
    offersInRadius,
    loadOffer,
    loadCityOffers,
    card,
    user,
    openAuthPopup,
    changeBookmark,
    popupModal,
  } = props;
  const { id } = props.match.params;
  const [isFavorite, setIsFavorite] = useState(false);

  useEffect(() => {
    loadOffer(id);
  }, [id]);

  useEffect(() => {
    if (card) {
      loadCityOffers(card.city.name);
    }
  }, [card]);

  useEffect(() => {
    if (card && user.bookmarks && user.bookmarks.indexOf(card._id) !== -1) {
      setIsFavorite(true);
    } else {
      setIsFavorite(false);
    }
  }, [card, user]);

  const bookmarkClickHandler = () => {
    if (isAuthorizationRequired) {
      openAuthPopup();
    } else {
      setIsFavorite(!isFavorite);
      changeBookmark(card._id);
    }
  };

  return (
    <div>
      <div style={{ display: "none" }}>
        <svg xmlns="http://www.w3.org/2000/svg">
          <symbol id="icon-arrow-select" viewBox="0 0 7 4">
            <path
              fillRule="evenodd"
              clipRule="evenodd"
              d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
            ></path>
          </symbol>
          <symbol id="icon-bookmark" viewBox="0 0 17 18">
            <path d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"></path>
          </symbol>
          <symbol id="icon-star" viewBox="0 0 13 12">
            <path
              fillRule="evenodd"
              clipRule="evenodd"
              d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
            ></path>
          </symbol>
        </svg>
      </div>

      <div className="page">
        <Header />

        {popupModal && <AuthModal />}

        {card && (
          <main className="page__main page__main--property">
            <section className="property">
              <div className="property__gallery-container container">
                <div className="property__gallery">
                  {card.images.map((image, i) => (
                    <div key={i} className="property__image-wrapper">
                      <img
                        className="property__image"
                        src={image}
                        alt="Photo studio"
                      />
                    </div>
                  ))}
                </div>
              </div>
              <div className="property__container container">
                <div className="property__wrapper">
                  <div
                    style={{ display: !card.is_premium && "none" }}
                    className="property__mark"
                  >
                    <span>Premium</span>
                  </div>
                  <div className="property__name-wrapper">
                    <h1 className="property__name">{card.title}</h1>
                    <button
                      className={`${
                        isFavorite && "property__bookmark-button--active "
                      } property__bookmark-button button`}
                      type="button"
                      onClick={() => bookmarkClickHandler()}
                    >
                      <svg
                        className="property__bookmark-icon"
                        width="31"
                        height="33"
                      >
                        <use xlinkHref="#icon-bookmark"></use>
                      </svg>
                      <span className="visually-hidden">To bookmarks</span>
                    </button>
                  </div>
                  <div className="property__rating rating">
                    <div className="property__stars rating__stars">
                      <span
                        style={{ width: `${(card.rating / 5) * 100}%` }}
                      ></span>
                      <span className="visually-hidden">Rating</span>
                    </div>
                    <span className="property__rating-value rating__value">
                      {card.rating}
                    </span>
                  </div>
                  <ul className="property__features">
                    <li className="property__feature property__feature--entire">
                      {card.type}
                    </li>
                    <li className="property__feature property__feature--bedrooms">
                      {card.bedrooms}{" "}
                      {card.bedrooms === 1 ? "Bedroom" : "Bedrooms"}
                    </li>
                    <li className="property__feature property__feature--adults">
                      Max {card.max_adults}{" "}
                      {card.max_adults === 1 ? "Adult" : "Adults"}
                    </li>
                  </ul>
                  <div className="property__price">
                    <b className="property__price-value">&euro;{card.price}</b>
                    <span className="property__price-text">&nbsp;night</span>
                  </div>
                  <div className="property__inside">
                    <h2 className="property__inside-title">
                      What&apos;s inside
                    </h2>
                    <ul className="property__inside-list">
                      {card.goods.map((item, i) => (
                        <li key={i} className="property__inside-item">
                          {item}
                        </li>
                      ))}
                    </ul>
                  </div>
                  <div className="property__host">
                    <h2 className="property__host-title">Meet the host</h2>
                    <div className="property__host-user user">
                      <div
                        className={`property__avatar-wrapper user__avatar-wrapper ${
                          card.host.is_pro && "property__avatar-wrapper--pro"
                        }`}
                      >
                        <img
                          className="property__avatar user__avatar"
                          src={`/${card.host.avatar_url}`}
                          width="74"
                          height="74"
                          alt="Host avatar"
                        />
                      </div>
                      <span className="property__user-name">
                        {card.host.name}
                      </span>
                      <span className="property__user-status">Pro</span>
                    </div>
                    <div className="property__description">
                      <p className="property__text">{card.description}</p>
                    </div>
                  </div>
                  {/* <section className="property__reviews reviews">
                    <Comments id={id} />
                    {!isAuthorizationRequired && <CommentForm />}
                  </section> */}
                </div>
              </div>
              <section className="property__map map">
                <Map renderCircle currentOffer={card} />
              </section>
            </section>
            <div className="container">
              <section className="near-places places">
                <h2 className="near-places__title">
                  Other places in the neighbourhood
                </h2>
                <div className="near-places__list places__list">
                  {offersInRadius &&
                    offersInRadius.map((offerCard) => (
                      <Card key={offerCard._id} card={offerCard} />
                    ))}
                </div>
              </section>
            </div>
          </main>
        )}
      </div>
    </div>
  );
};

OfferPage.propTypes = {
  loadOffer: PropTypes.func.isRequired,
  loadCityOffers: PropTypes.func.isRequired,
  openAuthPopup: PropTypes.func.isRequired,
  popupModal: PropTypes.func.isRequired,
  changeBookmark: PropTypes.func.isRequired,
  offersInRadius: PropTypes.arrayOf(
    PropTypes.shape({
      _id: PropTypes.string.isRequired,
      bedrooms: PropTypes.number.isRequired,
      city: PropTypes.shape({
        _id: PropTypes.string.isRequired,
        location: PropTypes.shape({
          latitude: PropTypes.number.isRequired,
          longitude: PropTypes.number.isRequired,
          zoom: PropTypes.number.isRequired,
        }),
        name: PropTypes.string.isRequired,
      }),
      description: PropTypes.string.isRequired,
      goods: PropTypes.array.isRequired,
      host: PropTypes.string.isRequired,
      images: PropTypes.array.isRequired,
      is_premium: PropTypes.bool.isRequired,
      location: PropTypes.shape({
        latitude: PropTypes.number.isRequired,
        longitude: PropTypes.number.isRequired,
        zoom: PropTypes.number.isRequired,
      }).isRequired,
      max_adults: PropTypes.number.isRequired,
      preview_image: PropTypes.string.isRequired,
      price: PropTypes.string.isRequired,
      rating: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      type: PropTypes.string.isRequired,
    }).isRequired
  ),
  card: PropTypes.shape({
    _id: PropTypes.string.isRequired,
    bedrooms: PropTypes.number.isRequired,
    city: PropTypes.shape({
      _id: PropTypes.string.isRequired,
      location: PropTypes.shape({
        latitude: PropTypes.number.isRequired,
        longitude: PropTypes.number.isRequired,
        zoom: PropTypes.number.isRequired,
      }),
      name: PropTypes.string.isRequired,
    }),
    description: PropTypes.string.isRequired,
    goods: PropTypes.array.isRequired,
    host: PropTypes.shape({
      _id: PropTypes.string.isRequired,
      avatar_url: PropTypes.string.isRequired,
      bookmarks: PropTypes.array.isRequired,
      email: PropTypes.string.isRequired,
      is_pro: PropTypes.bool.isRequired,
      name: PropTypes.string.isRequired,
      offers: PropTypes.array.isRequired,
    }).isRequired,
    images: PropTypes.array.isRequired,
    is_premium: PropTypes.bool.isRequired,
    location: PropTypes.shape({
      latitude: PropTypes.number.isRequired,
      longitude: PropTypes.number.isRequired,
      zoom: PropTypes.number.isRequired,
    }).isRequired,
    max_adults: PropTypes.number.isRequired,
    preview_image: PropTypes.string.isRequired,
    price: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
  }),
  user: PropTypes.shape({
    _id: PropTypes.string,
    avatar_url: PropTypes.string,
    bookmarks: PropTypes.array,
    email: PropTypes.string,
    is_pro: PropTypes.bool,
    name: PropTypes.string,
    offers: PropTypes.array,
  }),

  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
  isAuthorizationRequired: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) =>
  Object.assign({}, null, {
    isAuthorizationRequired: state.auth.isAuthorizationRequired,
    offersInRadius: state.offersInRadius.offersInRadius,
    card: state.offers.offer,
    user: state.user.user,
    popupModal: state.auth.popupModal,
  });

const mapDispatchToProps = (dispatch) => ({
  loadOffer: (id) => {
    dispatch(Operations.loadOffer(id));
  },
  loadCityOffers: (city) => {
    dispatch(Operations.loadCityOffers(city));
  },
  changeBookmark: (id) => {
    dispatch(UserOperations.changeBookmark(id));
  },
  openAuthPopup: () => {
    dispatch(ActionCreator.togglePopup());
  },
});

export { OfferPage };

export default connect(mapStateToProps, mapDispatchToProps)(OfferPage);
/* eslint-enable camelcase*/
