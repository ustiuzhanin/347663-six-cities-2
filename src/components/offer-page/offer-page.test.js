import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

import { offer as mockOffer } from "../../mocks/single-offer";
import { offerWithHost as mockOfferWithHost } from "../../mocks/single-offer";
import { user as mockUser } from "../../mocks/user";
import { OfferPage } from "./offer-page.jsx";

test(`OfferPage's Snapshot`, () => {
  const renderer = new ShallowRenderer();
  renderer.render(
    <OfferPage
      isAuthorizationRequired
      loadOffer={jest.fn()}
      loadCityOffers={jest.fn()}
      openAuthPopup={jest.fn()}
      popupModal={jest.fn()}
      changeBookmark={jest.fn()}
      offersInRadius={[mockOffer]}
      card={mockOfferWithHost}
      user={mockUser}
      match={{
        params: {
          id: "id12",
        },
      }}
    />
  );
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
