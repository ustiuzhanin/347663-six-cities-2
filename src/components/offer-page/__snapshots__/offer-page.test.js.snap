// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferPage's Snapshot 1`] = `
<div>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
    >
      <symbol
        id="icon-arrow-select"
        viewBox="0 0 7 4"
      >
        <path
          clipRule="evenodd"
          d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        id="icon-bookmark"
        viewBox="0 0 17 18"
      >
        <path
          d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
        />
      </symbol>
      <symbol
        id="icon-star"
        viewBox="0 0 13 12"
      >
        <path
          clipRule="evenodd"
          d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
          fillRule="evenodd"
        />
      </symbol>
    </svg>
  </div>
  <div
    className="page"
  >
    <Memo(Connect(Header)) />
    <Memo(Connect(AuthModal)) />
    <main
      className="page__main page__main--property"
    >
      <section
        className="property"
      >
        <div
          className="property__gallery-container container"
        >
          <div
            className="property__gallery"
          >
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/5.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/19.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/1.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/2.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/10.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/13.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/20.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/4.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/8.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/11.jpg"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/16.jpg"
              />
            </div>
          </div>
        </div>
        <div
          className="property__container container"
        >
          <div
            className="property__wrapper"
          >
            <div
              className="property__mark"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="property__name-wrapper"
            >
              <h1
                className="property__name"
              >
                Canal View Prinsengracht
              </h1>
              <button
                className="false property__bookmark-button button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="property__bookmark-icon"
                  height="33"
                  width="31"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="property__rating rating"
            >
              <div
                className="property__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "62%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
              <span
                className="property__rating-value rating__value"
              >
                3.1
              </span>
            </div>
            <ul
              className="property__features"
            >
              <li
                className="property__feature property__feature--entire"
              >
                room
              </li>
              <li
                className="property__feature property__feature--bedrooms"
              >
                1
                 
                Bedroom
              </li>
              <li
                className="property__feature property__feature--adults"
              >
                Max 
                2
                 
                Adults
              </li>
            </ul>
            <div
              className="property__price"
            >
              <b
                className="property__price-value"
              >
                €
                101
              </b>
              <span
                className="property__price-text"
              >
                 night
              </span>
            </div>
            <div
              className="property__inside"
            >
              <h2
                className="property__inside-title"
              >
                What's inside
              </h2>
              <ul
                className="property__inside-list"
              >
                <li
                  className="property__inside-item"
                >
                  Breakfast
                </li>
                <li
                  className="property__inside-item"
                >
                  Laptop friendly workspace
                </li>
              </ul>
            </div>
            <div
              className="property__host"
            >
              <h2
                className="property__host-title"
              >
                Meet the host
              </h2>
              <div
                className="property__host-user user"
              >
                <div
                  className="property__avatar-wrapper user__avatar-wrapper false"
                >
                  <img
                    alt="Host avatar"
                    className="property__avatar user__avatar"
                    height="74"
                    src="//img/avatar.svg"
                    width="74"
                  />
                </div>
                <span
                  className="property__user-name"
                >
                  test
                </span>
                <span
                  className="property__user-status"
                >
                  Pro
                </span>
              </div>
              <div
                className="property__description"
              >
                <p
                  className="property__text"
                >
                  This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.
                </p>
              </div>
            </div>
          </div>
        </div>
        <section
          className="property__map map"
        >
          <Memo(Connect(RenderMap))
            currentOffer={
              Object {
                "_id": "6005f36637751d35c8503011",
                "bedrooms": 1,
                "city": Object {
                  "_id": "6005f36637751d35c8503011",
                  "location": Object {
                    "latitude": 51.225402,
                    "longitude": 6.776314,
                    "zoom": 13,
                  },
                  "name": "Dusseldorf",
                },
                "description": "This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.",
                "goods": Array [
                  "Breakfast",
                  "Laptop friendly workspace",
                ],
                "host": Object {
                  "_id": "6005f36637751d35c8503011",
                  "avatar_url": "/img/avatar.svg",
                  "bookmarks": Array [
                    "5fee3aaa54b627023d7f94b7",
                  ],
                  "email": "test1@test.com",
                  "is_pro": false,
                  "name": "test",
                  "offers": Array [],
                },
                "images": Array [
                  "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/5.jpg",
                  "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/19.jpg",
                  "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/1.jpg",
                  "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/2.jpg",
                  "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/10.jpg",
                  "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/13.jpg",
                  "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/20.jpg",
                  "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg",
                  "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/4.jpg",
                  "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg",
                  "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/8.jpg",
                  "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg",
                  "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/11.jpg",
                  "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/16.jpg",
                ],
                "is_premium": false,
                "location": Object {
                  "latitude": 51.248402000000006,
                  "longitude": 6.763314,
                  "zoom": 16,
                },
                "max_adults": 2,
                "preview_image": "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg",
                "price": "101",
                "rating": 3.1,
                "title": "Canal View Prinsengracht",
                "type": "room",
              }
            }
            renderCircle={true}
          />
        </section>
      </section>
      <div
        className="container"
      >
        <section
          className="near-places places"
        >
          <h2
            className="near-places__title"
          >
            Other places in the neighbourhood
          </h2>
          <div
            className="near-places__list places__list"
          >
            <Memo(Connect(Card))
              card={
                Object {
                  "_id": "6005f36637751d35c8503011",
                  "bedrooms": 1,
                  "city": Object {
                    "_id": "6005f36637751d35c8503011",
                    "location": Object {
                      "latitude": 51.225402,
                      "longitude": 6.776314,
                      "zoom": 13,
                    },
                    "name": "Dusseldorf",
                  },
                  "description": "This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.",
                  "goods": Array [
                    "Breakfast",
                    "Laptop friendly workspace",
                  ],
                  "host": "6005f36637751d35c8503011",
                  "images": Array [
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/5.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/19.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/1.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/2.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/10.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/13.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/20.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/4.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/8.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/11.jpg",
                    "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/16.jpg",
                  ],
                  "is_premium": false,
                  "location": Object {
                    "latitude": 51.248402000000006,
                    "longitude": 6.763314,
                    "zoom": 16,
                  },
                  "max_adults": 2,
                  "preview_image": "https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg",
                  "price": "101",
                  "rating": 3.1,
                  "title": "Canal View Prinsengracht",
                  "type": "room",
                }
              }
            />
          </div>
        </section>
      </div>
    </main>
  </div>
</div>
`;
